### PR TITLE
Issue 3 view posts

### LINKED ISSUE #5: Issue 3 - User Can View Posts
- [ ] Create a components and pages folder in `client/src`
  - [ ] The components folder should have a `Header` and `SideBar` component
  - [ ] Create the `Posts` component for the user to view posts.
- [ ] `npm` install the Mantine UI React library in the `client/` folder
- [ ] Create mobile UI markup based on Figma
- [ ] Then create desktop UI markup

### PR DESCRIPTION
**Pull request by techmatlock**
Recording 2024-08-01 at 08 59 28

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,5 +1,62 @@
...
+        setIsMobile(true);
+      } else {
+        setIsMobile(false);
+      }
+    }
+    window.addEventListener('resize', onResize);
+    return () => {
+      window.removeEventListener('resize', onResize);
+    };
+  });
```
Comment (shawnkost) [inline client/src/App.tsx]: Missing dependency array

File: client/src/components/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            </li>
+            <li className="rounded-lg mx-4 mb-4 hover:bg-gray-200">
+              <Link to="#">Snacks</Link>
+            </li>
+            <li className="rounded-lg mx-4 mb-4 hover:bg-gray-200">
+              <Link to="#">Fast Food</Link>
+            </li>
+            <li className="rounded-lg mx-4 mb-4 hover:bg-gray-200">
+              <Link to="#">Ingredient Swaps</Link>
+            </li>
```
Comment (shawnkost) [inline client/src/components/NavBar.tsx]: Works, but don't you have an endpoint to get all of the categories? If you pull from the database, and map the results, it'd make it more dynamic

**Review by shawnkost — APPROVED** [review]
Looking great! Couple comments here

File: client/src/App.tsx (hunk 1 above)
Comment (techmatlock) [reply client/src/App.tsx]: Added

File: client/src/components/NavBar.tsx (hunk 2 above)
Comment (techmatlock) [reply client/src/components/NavBar.tsx]: Added

File: client/src/components/NavBar.tsx [hunk 3]
```diff
@@ -17,21 +22,13 @@ export function NavBar() {
...
-            </li>
-            <li className="rounded-lg mx-4 mb-4 hover:bg-gray-200">
-              <Link to="#">Fast Food</Link>
-            </li>
-            <li className="rounded-lg mx-4 mb-4 hover:bg-gray-200">
-              <Link to="#">Ingredient Swaps</Link>
-            </li>
+            {categories.map((category, index) => (
+              <li
+                key={category.name + index}
```
Comment (shawnkost) [inline client/src/components/NavBar.tsx]: I would use the `category.id` for the `key`, since the `id`'s will always be unique

**Pull request merged**